src/Microsoft.Extensions.Logging.AzureAppServices/Internal/FileLoggerProvider.cs

```
@@ -1,98 +1,80 @@
-            {
-                Directory.CreateDirectory(logsFolder);
-            }
-            var logsFilePattern = Path.Combine(logsFolder, FileNamePattern);
+            return Path.Combine(_path, $"{_fileName}{group.Year:0000}{group.Month:00}{group.Day:00}.txt");
+        }
 
-            var messageFormatter = new MessageTemplateTextFormatter(_outputTemplate, null);
-            var rollingFileSink = new RollingFileSink(logsFilePattern, messageFormatter, _fileSizeLimit, _retainedFileCountLimit);
+        public (int Year, int Month, int Day) GetGrouping(LogMessage message)
+        {
+            return (message.Timestamp.Year, message.Timestamp.Month, message.Timestamp.Day);
+        }
 
-            ILogEventSink flushingSink;
-            if (_flushPeriod != null)
-            {
-                flushingSink = new PeriodicFlushToDiskSink(rollingFileSink, _flushPeriod.Value);
-            }
-            else
+        protected void RollFiles()
+        {
+            if (_maxRetainedFiles > 0)
             {
-                flushingSink = rollingFileSink;
-            }
-            var backgroundSink = new BackgroundSink(flushingSink, _backgroundQueueSize);
+                var files = new DirectoryInfo(_path)
+                    .GetFiles(_fileName + "*")
+                    .OrderByDescending(f => f.Name)
+                    .Skip(_maxRetainedFiles.Value);
 
-            var loggerConfiguration = new LoggerConfiguration();
-            loggerConfiguration.WriteTo.Sink(backgroundSink);
-            loggerConfiguration.MinimumLevel.ControlledBy(new WebConfigurationReaderLevelSwitch(reader,
-                configuration =>
+                foreach (var item in files)
                 {
-                    return configuration.FileLoggingEnabled ? configuration.FileLoggingLevel : LogLevel.None;
-                }));
-            return loggerConfiguration.CreateLogger();
+                    item.Delete();
```

`WriteMessagesAsync` invocation is wrapped into try/catch that suppresses all exception.